lib/datasource-processor.js

```diff
@@ -459,15 +453,15 @@ function getBands(ds, bandCount){
         nodata = band.noDataValue;
         unitType = ds.bands.get(i).unitType;
         //Error check file sources
-        try { 
+        try {
             bandStats = band.getStatistics(false, true);
         } catch(err){
-            return invalid("Error getting statistics of band. 1 or more of the VRT file's relative sources may be missing: " + err.message);
-        };
+            throw invalid("Error getting statistics of band. 1 or more of the VRT file's relative sources may be missing: " + err.message);
+        }
```

Comment:
Yeah it actually is intentional, because this function [is called synchronously here](https://github.com/mapbox/mapnik-omnivore/blob/4134468b53257108724e8e994b9647ff4c3117c8/lib/datasource-processor.js#L406) and a conditional in the `catch` passes the message along.